ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_program_factory_rm.cpp

```diff
@@ -0,0 +1,205 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <vector>
+
+#include "fold_device_operation.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/operations/moreh/moreh_helper_functions.hpp"
+
+namespace ttnn::operations::moreh::moreh_fold {
+
+using namespace tt;
+using namespace tt::tt_metal;
+
+MorehFoldOperation::ProgramFactory::cached_program_t MorehFoldOperation::ProgramFactory::create(
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output) {
+    auto& input = tensor_args.input;
+
+    auto output_size = operation_attributes.output_size;
+    auto kernel_size = operation_attributes.kernel_size;
+    auto dilation = operation_attributes.dilation;
+    auto padding = operation_attributes.padding;
+    auto stride = operation_attributes.stride;
+    auto output_shape = output.get_logical_shape().as_vector();
+
+    uint32_t spatial_dim_size = output_shape.size() - 2;
+    uint32_t kernel_size_product = 1;
+    std::vector<uint32_t> ls;
+    uint32_t L = 1;
+    for (uint32_t i = 0; i < spatial_dim_size; ++i) {
+        uint32_t l =
+            (((operation_attributes.output_size[i] + 2 * operation_attributes.padding[i] -
+               operation_attributes.dilation[i] * (operation_attributes.kernel_size[i] - 1) - 1) /
+              operation_attributes.stride[i]) +
+             1);
```

Comment:
```suggestion
        uint32_t l =
            (((output_size[i] + 2 * padding[i] -
               dilation[i] * (kernel_size[i] - 1) - 1) /
              stride[i]) +
             1);
```
I think it would be better to modify it this way since the `output_size` variable is created above.

Reply:
Resolved in lastest commit